src/opencl/utils.rs

```diff
@@ -21,6 +21,18 @@ pub fn is_little_endian(d: ocl::Device) -> GPUResult<bool> {
     }
 }
 
+pub fn get_bus_id(d: ocl::Device) -> ocl::Result<u32> {
+    let vendor = d.vendor()?;
+    match vendor.as_str() {
```

Comment:
Safer would be to lower case the vendor and match against that, maybe even just a `contains`, to avoid having missing matches on small differences.

Reply:
I think it is case sensitive though. I think we really do want to treat it as an opaque identifier to be matched exactly.